src/ro/sci/h9/Main.java

```diff
@@ -0,0 +1,36 @@
+package ro.sci.h9;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * @author Emanuel Pruker
+ */
+public class Main {
+
+    public static void main(String[] args) {
+        Path inPath = Paths.get("src", "input.csv");
+        System.out.println("csv exists: " + Files.exists(inPath));
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(inPath.toAbsolutePath().toString()))) {
+
+            String line;
```

Comment:
This is the same as doing a readAll of the file in memory. There is no gain in using the buffered reader if the entire file is basically rebuilt in memory in the stringbuffer. Also the stringbuffer is basically never used as the first operation you do is to split it and put it in the treeset. So why not building the treeset when the lines are read instead of reading line by line then building a large string and the split to lines again.